source/common/http/filter_manager.cc

```diff
@@ -1489,6 +1511,27 @@ void FilterManager::maybeEndEncode(bool end_stream) {
   if (end_stream) {
     ASSERT(!state_.remote_encode_complete_);
     state_.remote_encode_complete_ = true;
+    if (allow_upstream_half_close_) {
+      maybeCloseStream();
```

Comment:
should this be half close?  closeUpstream?

should this logic all be in maybeClose?

Reply:
It closes downstream and upstream streams. I have renamed it `checkAndCloseStreamIfFullyClosed` to hopefully make it clearer.